ProcessLib/RichardsFlow/RichardsFlowMaterialProperties.h

```diff
@@ -0,0 +1,121 @@
+/**
+* \copyright
+* Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+*            Distributed under a Modified BSD License.
+*              See accompanying file LICENSE.txt or
+*              http://www.opengeosys.org/project/license
+*
+*/
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "MaterialLib/Fluid/FluidPropertyHeaders.h"
+#include "MaterialLib/Fluid/FluidProperties/FluidProperties.h"
+#include "MaterialLib/PhysicalConstant.h"
+#include "MaterialLib/PorousMedium/Porosity/Porosity.h"
+#include "MaterialLib/PorousMedium/PorousPropertyHeaders.h"
+#include "MaterialLib/PorousMedium/Storage/Storage.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CapillaryPressureSaturation.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CreateCapillaryPressureModel.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/RelativePermeability/CreateRelativePermeabilityModel.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/RelativePermeability/RelativePermeability.h"
+
+namespace MeshLib
+{
+template <typename PROP_VAL_TYPE>
+class PropertyVector;
+}
+
+namespace ProcessLib
+{
+class SpatialPosition;
+namespace RichardsFlow
+{
+/** This class has a collection of material properties for Richards flow model
+*  and it makes description of the properties of unsaturated porous media
+*  i.e. the fluid density and viscosity models
+*  the relative permeability models,
+*  the capillary pressure-saturation relationships.
+*  It generally provides the computation of the PDE coefficients for Richards
+* flow.
+*/
+
+class RichardsFlowMaterialProperties final
```

Comment:
Future work could be to create a class for single phase fluid flow properties, and it has common fluid property data for LiquidFlow, GroundWaterFlow and RichardsFlow.